### ISSUE TITLE
Performance sensitive analyzers package

### ISSUE THREAD

**Issue by Youssef1313** [dotnet/roslyn-analyzers#4721]
I'm having some questions (and one suggestion) regarding this package:

- (suggestion) The package currently supports C# only and is based on syntax analysis. It could be improved to support VB using operation analysis, or add VB syntax analysis as well.
- Currently all rules are in `AnalyzerReleases.Unshipped`, is there any work to release it?
- It appears that this package was a direct port from <https://github.com/microsoft/RoslynClrHeapAllocationAnalyzer> even with the same diagnostic ids. So, when releasing, should it use different diagnostic ids? I'm not sure what will happen if a user has a reference to the one in the other repo and this one?

**Comment by mavasani** [comment]
Tagging @genlu @sharwell 

IMO, this is still an experimental package, with no concrete plans at the moment towards a release.

**Comment by Youssef1313** [comment]
@mavasani Is it worth attempting to re-write these analyzers as operation analyzers for now? or let this task when some plans regarding release are made?

**Comment by mavasani** [comment]
@Youssef1313 We have https://github.com/dotnet/roslyn-analyzers/pull/3016 and https://github.com/dotnet/roslyn-analyzers/pull/2228 to already kick start that effort. Would need someone to help review/drive those PRs to start with.

**Comment by genlu** [comment]
>(suggestion) The package currently supports C# only and is based on syntax analysis. It could be improved to support VB using operation analysis, or add VB syntax analysis as well.

Yes, this is tracked by an umbrella issue https://github.com/dotnet/roslyn-analyzers/issues/2120

> So, when releasing, should it use different diagnostic ids?

Tracked by https://github.com/dotnet/roslyn-analyzers/issues/2259. I'm inclining to have new IDs (but not with prefix "CA", probably "PS"). @sharwell thoughts?

>  is there any work to release it?

One thing that blocks the publication of this package is editorconfig support. When I ported it over from `RoslynClrHeapAllocationAnalyzer`, I made the analysis conditioned on the presence of an attribute, since which is our main use case (in Roslyn repo). To make it behave the same as the original (therefore useful for most users), we need an option to say "analyzing annotated symbols only or everything". Once we have this in place, I think we will be able release it.

**Comment by Youssef1313** [comment]
> we need an option to say "analyzing annotated symbols only or everything". Once we have this in place, I think we will be able release it.

I'll try to look into that.

**Comment by genlu** [comment]
@Youssef1313 That'd be awesome! Thanks!

**Comment by dominikjeske** [comment]
I have a question about this package. On this site https://github.com/Microsoft/RoslynClrHeapAllocationAnalyzer there is information about moving to dotnet/roslyn-analyzers. Are there any work in this area? I can see some 3.3.5 beta but issue is not a part of any milestone so I'm wonder what is going on with this topic?
